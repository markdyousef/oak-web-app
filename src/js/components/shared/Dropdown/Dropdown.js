// @flow
import React, { PropTypes } from 'react';
import styled from 'styled-components';

const Arrow = styled.div`
    height: 14px;
    width: 14px;
    border-top: 1px solid #E5E5E5;
    border-left: 1px solid #E5E5E5;
    background: var(--white);
    position: absolute;
    top: -7px;
    transform: rotate(45deg);
    background-color: #fff;
`;

const Menu = styled.div`
    display: block;
    box-shadow: 1px 3px 7px rgba(32, 33, 35, 0.05);
    width: 100%;
    background-color: #fff;
    border: 1px solid #efefef;
    border-radius: 3px;
    padding: 20px;
    & a {
        color: #000;
        display: block;
        margin-top: 10px;
        font-size: 16px;
        font-family: 'Proxima Nova'
    }
`;

const Dropdown = ({ children, arrowPos }:Object) => {
    let style;
    switch (arrowPos) {
    case 'left':
        style = { left: '25px' };
        break;
    default:
        style = { right: '25px' };
        break;
    }
    return (
        <div>
            <Arrow style={style} />
            <Menu>
                {children}
            </Menu>
        </div>
    );
};

Dropdown.propTypes = {
    children: PropTypes.node,
    arrowPos: PropTypes.string
};

Dropdown.defaultProps = {
    children: null,
    arrowPos: null
};

export default Dropdown;
