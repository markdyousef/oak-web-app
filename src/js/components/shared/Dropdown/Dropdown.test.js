// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import Dropdown from './Dropdown';

test('renders Dropdown', () => {
    const props = {
    };
    const component = renderer.create(
        <Dropdown {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
