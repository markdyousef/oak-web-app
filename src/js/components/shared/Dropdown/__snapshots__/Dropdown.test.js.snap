exports[`test renders Dropdown 1`] = `
<div>
  <div
    className="hqCZJo"
    style={
      Object {
        "right": "25px",
      }
    } />
  <div
    className="fEcnKi" />
</div>
`;
